src/porepy/models/poromechanics.py

```diff
@@ -218,33 +201,35 @@ def update_time_dependent_ad_arrays(self, initial: bool) -> None:
         # Call super in case class is combined with other classes implementing this
         # method.
         super().update_time_dependent_ad_arrays(initial)
-        # Update the mechanical boundary conditions to both the solutions and iterates.
-        for sd, data in self.mdg.subdomains(return_data=True, dim=self.nd):
-            if initial:
-                vals = self.time_dependent_bc_values_mechanics([sd])
-                pp.set_solution_values(
-                    name=self.bc_values_mechanics_key,
-                    values=vals,
-                    data=data,
-                    time_step_index=0,
-                )
-            else:
-                # Copy old values from iterate to the solution.
-                vals = data[pp.ITERATE_SOLUTIONS][self.bc_values_mechanics_key][0]
-                pp.set_solution_values(
-                    name=self.bc_values_mechanics_key,
-                    values=vals,
-                    data=data,
-                    time_step_index=0,
-                )
-
-            vals = self.time_dependent_bc_values_mechanics([sd])
-            pp.set_solution_values(
-                name=self.bc_values_mechanics_key,
-                values=vals,
-                data=data,
-                iterate_index=0,
-            )
+        assert False, "Assert we can remove this function"
```

Comment:
Is there surely no other use for this then updating BC values?

Reply:
Good question. Am I right that the reason you propose to delete this one is because it should be handled in a general method (not specific to poromechanics), @Yuriyzabegaev?

Reply:
@IvarStefansson @vlipovac yes, at least I don't see if it does something more than that. I just tried to be careful with changing the logic of the poromechanics setup.

Reply:
Fixed.